### PR TITLE
ZENKO-2554 VaultClient in Go - DeleteAccount

### PR DESCRIPTION
**Pull request by nicolas2bert**

### CODE REVIEW

**Review by alexanderchan-scality — APPROVED** [review]

File: golang/vaultclient/deleteAccount.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+		HTTPPath:   "/",
+	}
+
+	if input == nil {
+		input = &DeleteAccountInput{}
+	}
+
+	output = &DeleteAccountOutput{}
+	req = c.newRequest(op, input, output)
+	return
```
Comment (miniscruff) [inline golang/vaultclient/deleteAccount.go:68]: new to go but do you need this return here?

File: golang/vaultclient/deleteAccount.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+// DeleteAccountOutput contains the response to a successful DeleteAccount request.
+type DeleteAccountOutput struct{}
```
Comment (miniscruff) [inline golang/vaultclient/deleteAccount.go:72]: new to go, is it common to create empty structs like this? Or am I missing something?

File: golang/vaultclient/deleteAccount.go (hunk 1 above)
Comment (nicolas2bert) [reply golang/vaultclient/deleteAccount.go:68]: It is really just a shorthand for `return req, output`

File: golang/vaultclient/deleteAccount.go (hunk 2 above)
Comment (nicolas2bert) [reply golang/vaultclient/deleteAccount.go:72]: yes, the response is supposed to be empty.

**Review by miniscruff — APPROVED** [review]
approving but not a go expert

**Comment by nicolas2bert** [comment]
@bert-e approve

**Review by rahulreddy — APPROVED** [review]

**Comment by nicolas2bert** [comment]
@bert-e approve

**Pull request merged**